### PR TITLE
Fix MG_test bug

### LINKED ISSUE #1706: [BUG] Test_mg_batch_betweenness_centrality returns a type error
test_mg_batch_betweenness centrality returns the following type error 
`distributed.worker - WARNING - Compute Failed
Function:  _replicate_cudf_series
args:      ((None, 156, dtype('int32')), b'\x079i\x11D\xb0AB\x94\xcf\xd1j[`e2')
kwargs:    {}
Exception: TypeError("'NoneType' object is not subscriptable")`

One of the workers has no data `odict_items([('tcp://127.0.0.1:35867', [<Future: finished, type: cudf.Series, key: ('from_cudf-946854107b6da760966a5c6127cfda4b', 0)>]), ('tcp://127.0.0.1:45685', None)])` and the variable(supposed to be a list) containing the data will be index later which returns the above TypeError

### LINKED ISSUE #1715: [BUG] Test_mg_renumber fails with TypeError
test_mg_renumber returns the following TypeError when using 3, 4, 5, 6, 7, 8 GPUs on DGX18

`Function:  call_renumber
args:      (b'\xa41qp\x05\xcbH\x90\xae\x9e\x02\x9d\x89\xb6\xffa', None, 312, True, False)
kwargs:    {}
Exception: TypeError("'NoneType' object is not subscriptable")`

### LINKED ISSUE #1716: [BUG] Test_mg_replication fails with a TypeError
Test_mg_replication returns the following type error:

`distributed.worker - WARNING - Compute Failed
Function:  _replicate_cudf_dataframe
args:      ((None, array(['src', 'dst', 'value'], dtype=object), src        int32
dst        int32
value    float32
dtype: object, 318), b"qo\x91'e\x1fN\x1c\xbd\xa0\x1e\xb7P\x8aF\xf8")
kwargs:    {}
Exception: TypeError("'NoneType' object is not subscriptable")`

### PR DESCRIPTION
**Pull request by jnke2016**
Fix `test_mg_batch_betweenness_centrality`, `test_mg_batch_edge_betweenness_centrality`, `test_mg_replication`, `test_mg_renumber`

closes #1706 
closes #1715 
closes #1716

### CODE REVIEW

File: python/cugraph/dask/common/input_utils.py [hunk 1]
```diff
@@ -196,6 +197,10 @@ def _workers_to_parts(futures):
         if w_to_p_map[w] is None:
             w_to_p_map[w] = []
         w_to_p_map[w].append(p)
+    _w_to_p_map = w_to_p_map.copy()
+    for w, p in _w_to_p_map.items():
+        if p is None:
+            del w_to_p_map[w]
```
Comment (rlratzel) [inline python/cugraph/dask/common/input_utils.py:203]: I don't know if this is any faster, but it prevents having to copy the entire dictionary:
```suggestion
    keys_to_delete = [w for (w,p) in w_to_p_map.items() if p is None]
    for k in keys_to_delete:
        del w_to_p_map[k]
```

**Review by rlratzel — COMMENTED** [review]
Looks good, I just had a few suggestions.

File: python/cugraph/dask/common/input_utils.py (hunk 1 above)
Comment (jnke2016) [reply python/cugraph/dask/common/input_utils.py:203]: A list comprehension to get the keys that have None values looks better and now I am only iterating over the keys to delete.

**Review by rlratzel — APPROVED** [review]

**Review by BradReesWork — APPROVED** [review]

**Comment by BradReesWork** [comment]
@gpucibot merge

**Pull request merged**